app/controllers/adopter_waitlists_controller.rb

```diff
@@ -0,0 +1,70 @@
+class AdopterWaitlistsController < ApplicationController
+  before_action :edit_my_adopters_user, only: %i(create update destroy)
+  before_action :edit_all_adopters_user, only: %i(create update destroy)
+  before_action :load_waitlist, only: %i(create)
+  before_action :load_adopter_waitlist, only: %i(show update destroy)
+  before_action :load_adopter, only: %i(create)
+
+  respond_to :html, :json
+
+  def index
+    redirect_to :root
+  end
+
+  def create
+    @adopter_waitlist = AdopterWaitlist.find_or_initialize_by(create_params)
+
+    if AdopterWaitlist.count == 0
```

Comment:
I think you want `if  @adopter_waitlist.persisted?` here, you just want to see if it exists already right?  If it's brand new default the rank to 1?

Reply:
Actually this whole conditional can be easily done with

```
@adopter_waitlist = AdopterWaitlist.where(waitlist: @waitlist).count + 1
```

Maybe we put this in a `before_save` on the `AdopterWaitlist` model and only execute if rank is nil?

Reply:
I was trying to add an adopter at the end of the waitlist. If currently there is no adopter, set rank to 1. Maybe I can use `@adopter_waitlist.rank = AdopterWaitlist.where(waitlist: @waitlist).count + 1` instead of `@adopter_waitlist.rank = AdopterWaitlist.where(wailist: @waitlist).maximum('Rank') + 1`?